src/Model/ModelManager.php

```diff
@@ -339,16 +339,10 @@ public function getIdentifierValues($entity)
                 continue;
             }
 
-            if (method_exists($value, '__toString')) {
-                $identifiers[] = (string) $value;
-
-                continue;
-            }
-
             $fieldType = $metadata->getTypeOfField($name);
             $type = $fieldType && Type::hasType($fieldType) ? Type::getType($fieldType) : null;
             if ($type) {
-                $identifiers[] = $type->convertToDatabaseValue($value, $platform);
+                $identifiers[] = 'binary' === $platform->getDoctrineTypeMapping($fieldType) ? (string) $type->convertToPHPValue($value, $platform) : $type->convertToDatabaseValue($value, $platform);
```

Comment:
Please remove the ternary operator here and use a simple if statement, because this line is too long for a quick understanding

Reply:
I think that in this way it is a little confuse, but @phansys ask for that change. Please, could you two agree?

Reply:
I think what @core23 is proposing is different of how it was originally, consider this:
```php
if ($type) {
    if ('binary' === $platform->getDoctrineTypeMapping($fieldType)) {
        $identifiers[] = ? (string) $type->convertToPHPValue($value, $platform);
    } else {
        $identifiers[] = $type->convertToDatabaseValue($value, $platform);
    }
}
```

Reply:
With that approach there are too many if blocks nested. But if you think that it is better, ok ;-)

Reply:
Regardless the nested `if`s, personally the original approach looks more confusing to me since both `if` and `elseif` conditions were checking against `$type` evaluating to `true`.
Whatever, I'll approve the PR since the CS issues were fixed :+1:

Reply:
We will wait for the @core23 opinion to make the fix.

Reply:
We should not write lines that are longer than 80/120 chars (@greg0ire  😅 ).

Maybe the best way is a separate method that will return the correct identifiert if any other way is too confusing.

Reply:
Ok, I was thinking about that and I propose: 
```suggestion
                $identifiers[] = $this->getValueFromType($value, $type, $fieldType, $platform);
```
With this function:
```suggestion
    /**
     * 
     * @param mix $value
     * @param Doctrine\DBAL\Types\Type $type
     * @param string $fieldType
     * @param Doctrine\DBAL\Platforms\AbstractPlatform $platform
     */
    private function getValueFromType ($value, $type, $fieldType, $platform) 
    {
        if ('binary' === $platform->getDoctrineTypeMapping($fieldType)){
            return (string) $type->convertToPHPValue($value, $platform);
        }else {
            return $type->convertToDatabaseValue($value, $platform);
        } 
    }
```
What do you think?

Reply:
I'm ok with this if you use real type declarations where possible and if the superfluous `else` is removed:
```php
    /**
     * @param mixed $value
     * 
     * @return mixed
     */
    private function getValueFromType($value, Type $type, string $fieldType, AbstractPlatform $platform) 
    {
        if ('binary' === $platform->getDoctrineTypeMapping($fieldType)){
            return (string) $type->convertToPHPValue($value, $platform);
        }

        return $type->convertToDatabaseValue($value, $platform); 
    }

Reply:
Done